crates/virtio-queue/src/state.rs

```diff
@@ -426,15 +419,39 @@ impl QueueStateT for QueueState {
         self.next_avail.0
     }
 
-    fn next_used(&self) -> u16 {
-        self.next_used.0
-    }
-
     fn set_next_avail(&mut self, next_avail: u16) {
         self.next_avail = Wrapping(next_avail);
     }
 
+    fn next_used(&self) -> u16 {
+        self.next_used.0
+    }
+
     fn set_next_used(&mut self, next_used: u16) {
         self.next_used = Wrapping(next_used);
     }
+
+    fn pop_descriptor_chain<M>(&mut self, mem: M) -> Option<DescriptorChain<M>>
+    where
+        M: Clone + Deref,
+        M::Target: GuestMemory,
+    {
+        // Default, iter-based impl. Can most likely be improved.
+        self.iter(mem).ok()?.next()
```

Comment:
Should we for now log an error here instead of completely disregarding the error?

Reply:
sure